### PR TITLE
Added metrics for Mesos slaves and tasks running on them.

### PR DESCRIPTION
**Pull request by harnash**
### Required for all PRs:

- [X] CHANGELOG.md updated
- [X] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [X] README.md updated (if adding a new plugin)

This PR will allow gathering metrics from Mesos slaves along with metrics for each task running on them (optional). I had to change the configuration to allow gathering metrics from either master/slave which can be deployed on separate nodes (so no default for `masters` setting).
I've decided to put slave metrics in the same measurement `mesos` as master (sadly their names differ) but I've also added a tag which specifies which role a given metric belongs to (master/slave). Let me know if that is OK.
Task specific metrics will go to `mesos-tasks` measurement.

pinging the original author of the plugin: @tripledes

### CODE REVIEW

**Comment by sparrc** [comment]
@tripledes @asdfsx can you review?

**Comment by sparrc** [comment]
@harnash since it's a large change please remove the changelog update for now, because this will not be going into the latest release.

**Comment by harnash** [comment]
@sparrc Done!

**Comment by sparrc** [comment]
@tripledes @asdfsx Do you think you can review this? You don't need to review the code maybe just test it out. If it helps, there is a linux 64-bit binary available here: https://3818-33258973-gh.circle-artifacts.com/0//tmp/circle-artifacts.vJctDin/telegraf.gz

which you should be able to unpack and drop in-place on any linux system.

File: plugins/inputs/mesos/mesos.go [hunk 1]
```diff
@@ -17,33 +17,57 @@ import (
 	jsonparser "github.com/influxdata/telegraf/plugins/parsers/json"
 )
 
+type Role string
```
Comment (sparrc) [inline plugins/inputs/mesos/mesos.go]: why did you add this `Role` type?

Comment (harnash) [inline plugins/inputs/mesos/mesos.go]: Metrics collected from master (the one gathered so far) differs dramatically from the one collected from slaves. I use this to distinguish sets of metrics which are being collected by this plugin. 
As you can see below there are some functions which operates using this values to determine in which context they are being invoked. 
I hope that answers your question.

Comment (sparrc) [inline plugins/inputs/mesos/mesos.go]: Could you create a more specific type then? ie, create SlaveRole & MasterRole

Comment (harnash) [inline plugins/inputs/mesos/mesos.go]: @sparrc:
I'm not sure I understand you correctly. You want me to create two separate types like this:
```go
type SlaveRole string
type MasterRole string
```
I did implement one type (based on string) to allow easy mapping between this constant and string (used when specifying tags). If that is the problem I could refactor it to use `iota` and get rid of custom types at all.

Comment (sparrc) [inline plugins/inputs/mesos/mesos.go]: sorry, I think I've misunderstood it, disregard what I said before

**Comment by sparrc** [comment]
thanks much @harnash

**Comment by zstyblik** [comment]
@sparrc, @tripledes has moved on.
